File: src/main/java/ru/ioleksiv/telegram/bot/api/model/method/sticker/GetStickerSet.java
package ru.ioleksiv.telegram.bot.api.model.method.sticker;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.api.model.NetworkError;
import ru.ioleksiv.telegram.bot.api.model.objects.std.sticker.StickerSet;
import ru.ioleksiv.telegram.bot.core.controller.network.ActionNetworker;
import ru.ioleksiv.telegram.bot.core.model.method.RunnableAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#getstickerset>getStickerSet</a>
 */
public class GetStickerSet extends RunnableAction<StickerSet> {
    private static final String METHOD = "getStickerSet";

    /**
     * name	String	Yes	Name of the sticker set
     */
    @JsonProperty("name")
    private String name = null;

    public GetStickerSet(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    public String getName() {
        return name;
    }

    public GetStickerSet setName(String name) {
        this.name = name;
        return this;
    }

    @Override
    protected Class<? extends CommonResponse<StickerSet>> getResultWrapperClass() {
        return ResponseCollection.StickerSetResponse.class;
    }

    @Override
    public GetStickerSet setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/sticker/GetStickerSet.kt
package io.github.oleksivio.tl.kbot.server.api.method.sticker

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.StickerSetResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.Action
import io.github.oleksivio.tl.kbot.server.api.objects.std.sticker.StickerSet

/**
 * @see [getStickerSet](https://core.telegram.org/bots/api/#getstickerset)
 */
data class GetStickerSet(
    /**
     * name String Yes Name of the sticker set
     */
    @JsonProperty(ApiDict.NAME_KEY)
    val name: String
) : Action<StickerSet>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getStickerSet"

    @JsonIgnore
    override val resultWrapperClass = StickerSetResponse::class
}
